src/Mono.Android/Microsoft.Android.Runtime/ManagedValueManager.cs

```diff
@@ -251,13 +257,34 @@ public override List<JniSurfacedPeerInfo> GetSurfacedPeers ()
 			var peers = new List<JniSurfacedPeerInfo> (RegisteredInstances.Count);
 			foreach (var e in RegisteredInstances) {
 				foreach (var p in e.Value) {
-					peers.Add (new JniSurfacedPeerInfo (e.Key, new WeakReference<IJavaPeerable> (p)));
+					if (p.Target is not IJavaPeerable peer)
+						continue;
+					peers.Add (new JniSurfacedPeerInfo (e.Key, new WeakReference<IJavaPeerable> (peer)));
 				}
 			}
 			return peers;
 		}
 	}
 
+	static GCHandle CreateReferenceTrackingHandle (IJavaPeerable value) =>
+		JavaMarshal.CreateReferenceTrackingHandle (value, value.PeerReference.Handle);
```

Comment:
Started implementing the `JniObjectInfo` struct here:
* https://github.com/dotnet/java-interop/compare/dev/peppers/gcbridge

But we need `IJavaPeerable` to have some way to return the `IntPtr JniObjectInfo` so it can be used somehow like:
```csharp
static GCHandle CreateReferenceTrackingHandle (IJavaPeerable value) =>
	JavaMarshal.CreateReferenceTrackingHandle (value, value.JniObjectInfo);
```

Reply:
Now using `IJavaPeerable.JniObjectReferenceControlBlock`.